dists/darwin/build

```diff
@@ -12,60 +15,82 @@
 set -e
 
 # Software versions
-LIBUSB_VERSION=1.0.22
+LIBUSB_VERSION=1.0.24
+
+# Determine OS
+OS=darwin
 
 # Determine directories
 cd "$(dirname $0)/../.."
 PROJECT_DIR="$(pwd)"
 TARGET_DIR="$PROJECT_DIR/target"
-BUILD_DIR="$TARGET_DIR/build/darwin-x86-64"
-DOWNLOAD_DIR="$BUILD_DIR/downloads"
-ROOT_DIR="$BUILD_DIR/root"
+JAR_FILE=$TARGET_DIR/libusb4java-$OS.jar
+
+# Download libusb from GitHub and unzip
+DOWNLOAD_DIR="$TARGET_DIR/downloads"
 
 # Clean up target directory
-rm -rf "$BUILD_DIR"
+rm -rf "$TARGET_DIR"
 
 # Create download directory if not already present
 mkdir -p "$DOWNLOAD_DIR"
 
-# Determine OS and architecture
-OS=darwin
-ARCH=x86-64
-echo "Building for platform $OS-$ARCH"
-
-# Standard compiler and linker flags
-CFLAGS="-I$ROOT_DIR/include -arch x86_64"
-LDFLAGS="-L$ROOT_DIR/lib"
-
-# Export compiler and linker flags
-export CFLAGS LDFLAGS
-
-# Download and build libusb
-LIBUSB_TARBALL="libusb-$LIBUSB_VERSION.tar.bz2"
-LIBUSB_SOURCE="http://downloads.sf.net/project/libusb/libusb-1.0/libusb-$LIBUSB_VERSION/$LIBUSB_TARBALL"
+LIBUSB_TARBALL="v$LIBUSB_VERSION.zip"
+LIBUSB_SOURCE=https://github.com/libusb/libusb/archive/$LIBUSB_TARBALL
 LIBUSB_TARGET="$DOWNLOAD_DIR/$LIBUSB_TARBALL"
 if [ ! -f "$LIBUSB_TARGET" ]
 then
     curl -C - -o "$LIBUSB_TARGET.download" -L "$LIBUSB_SOURCE"
     mv -f "$LIBUSB_TARGET.download" "$LIBUSB_TARGET"
 fi
-mkdir -p "$BUILD_DIR/libusb"
-cd "$BUILD_DIR/libusb"
-tar xvf "$LIBUSB_TARGET" --strip-components=1
-./configure --disable-shared --enable-static --with-pic --prefix="$ROOT_DIR"
-make install-strip
-
-# Build libusb4java
-mkdir -p "$BUILD_DIR/libusb4java"
-cd "$BUILD_DIR/libusb4java"
-PKG_CONFIG_PATH="$ROOT_DIR/lib/pkgconfig" cmake "$PROJECT_DIR" \
-    -DCMAKE_BUILD_TYPE=Release \
-    -DCMAKE_INSTALL_PREFIX="" \
-    -DCMAKE_OSX_ARCHITECTURES=x86_64 \
-    -DLibUsb_USE_STATIC_LIBS=true
-make install/strip DESTDIR="$ROOT_DIR"
-
-# Create the JAR file
-mkdir -p "$BUILD_DIR/classes/org/usb4java/$OS-$ARCH"
-cp "$ROOT_DIR/lib/libusb4java.dylib" "$BUILD_DIR/classes/org/usb4java/$OS-$ARCH"
-jar cf "$TARGET_DIR/libusb4java-$OS-$ARCH.jar" -C "$BUILD_DIR/classes" org
+
+build_architecture () {
+  
+    local ARCH_ID=$1
+    local ARCH=$2
+    echo "Building for platform $OS-$ARCH"
+    local BUILD_DIR="$TARGET_DIR/build/darwin-$ARCH"
+    local ROOT_DIR="$BUILD_DIR/root"
+
+    # Standard compiler and linker flags
+    local CFLAGS="-I$ROOT_DIR/include -arch $ARCH"
+    local LDFLAGS="-L$ROOT_DIR/lib"
+
+    # Export compiler and linker flags
+    export CFLAGS LDFLAGS
+
+    # Unzip libusb
+    mkdir -p "$BUILD_DIR"
+    cd "$BUILD_DIR"
+    unzip "$LIBUSB_TARGET"
+    mv "$BUILD_DIR/libusb-$LIBUSB_VERSION" "$BUILD_DIR/libusb"
+    cd "$BUILD_DIR/libusb"
+
+    # Build libusb
+    autoreconf -i
+    ./configure --disable-shared --enable-static --with-pic --prefix="$ROOT_DIR" --host=$ARCH
```

Comment:
confiture's `host` option should contain `-darwin`.
No `-darwin` will choose null backend of libusb. As a result, usb4java returns no device.
https://github.com/libusb/libusb/blob/v1.0.26/configure.ac#L76